### PR TITLE
Update the release documentation to the new PROW CI.

### PR DESCRIPTION
**Pull request by awels**
Signed-off-by: Alexander Wels <awels@redhat.com>

**What this PR does / why we need it**:
We recently changed from using Travis CI to PROW for our releases. This PR updates the documentation on how to make a release.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: doc/releases.md [hunk 1]
```diff
@@ -31,46 +31,48 @@ When creating a new release branch, follow the below process.  This assumes that
 
     `$ git fetch <upstream>`
 
-1. Checkout the release branch locally
+1. Checkout and create the release branch locally
```
Comment (arnongilboa) [inline doc/releases.md:34]: Create and checkout

File: doc/releases.md [hunk 2]
```diff
@@ -31,46 +31,48 @@ When creating a new release branch, follow the below process.  This assumes that
...
-##### Images
+## Images
 
--  Check hub.docker.com/r/kubevirt repository for the newly tagged images. If you do not see the tags corresponding to the version, check the travis build log for errors.
+Ensure that the new images are available in quay.io/repository/kubevirt/container-name?tab=tags and that the version you specified in the tag is available
 
-   [CDI-Controller](https://hub.docker.com/r/kubevirt/cdi-controller/tags/)
+* [CDI controller](https://quay.io/repository/kubevirt/cdi-controller?tab=tags)
+* [CDI importer](https://quay.io/repository/kubevirt/cdi-importer?tab=tags)
+* [CDI clones](https://quay.io/repository/kubevirt/cdi-cloner?tab=tags)
```
Comment (arnongilboa) [inline doc/releases.md:71]: CDI cloner

File: doc/releases.md (hunk 1 above)
Comment (awels) [reply doc/releases.md:34]: Done

File: doc/releases.md (hunk 2 above)
Comment (awels) [reply doc/releases.md:71]: Done

**Pull request merged**